lib/src/reboot.rs

```diff
@@ -6,16 +6,29 @@ use fn_error_context::context;
 
 use crate::task::Task;
 
-/// Initiate a system reboot.
+/// Common reboot implementation that handles flushing streams and running the reboot command.
 /// This function will only return in case of error.
-#[context("Initiating reboot")]
-pub(crate) fn reboot() -> anyhow::Result<()> {
+fn perform_reboot(task_name: &str, command: &str) -> anyhow::Result<()> {
     // Flush output streams
     let _ = std::io::stdout().flush();
     let _ = std::io::stderr().flush();
-    Task::new("Rebooting system", "reboot").run()?;
-    tracing::debug!("Initiated reboot, sleeping forever...");
+    Task::new(task_name, command).run()?;
+    tracing::debug!("Initiated {}, sleeping forever...", task_name.to_lowercase());
     loop {
         std::thread::park();
     }
 }
+
+/// Initiate a system reboot.
+/// This function will only return in case of error.
+#[context("Initiating reboot")]
+pub(crate) fn reboot() -> anyhow::Result<()> {
+    perform_reboot("Rebooting system", "reboot")
+}
+
+/// Initiate a soft reboot (userspace-only restart).
+/// This function will only return in case of error.
+#[context("Initiating soft reboot")]
+pub(crate) fn soft_reboot() -> anyhow::Result<()> {
+    perform_reboot("Performing soft-reboot", "systemctl soft-reboot")
```

Comment:
We shouldn't use commands-as-strings (as would be parsed by a shell etc). I think the cleanest fix here is to change `perform_reboot` to take `enum RebootType { Hard, Soft }` or so, and then it invokes the right command